api/manifest/manifest.go

```diff
@@ -130,8 +135,11 @@ func Read(dir, filename string) (*Manifest, error) {
 				}
 			}
 
-			m[name].Volumes[i] = strings.Join(parts, ":")
+			entry.Volumes[i] = strings.Join(parts, ":")
 		}
+
+		entry.Manifest = &m
```

Comment:
The back-reference seems like it could lead to bugs in the future. Is there any way to get this by passing a manifest around where you need it?